src/access/acp.test.ts

```diff
@@ -20,349 +20,386 @@
  */
 
 import { describe, it } from "@jest/globals";
+import { WithAccessibleAcr } from "../acp/acp";
 import { AccessControlResource } from "../acp/control";
-import { addMockAcrTo, mockAcrFor } from "../acp/mock";
-import { createPolicy, setPolicy } from "../acp/policy";
-import { acp, rdf } from "../constants";
-import { UrlString, WithServerResourceInfo } from "../interfaces";
+import { internal_createControl } from "../acp/control.internal";
+import { addMockAcrTo } from "../acp/mock";
+import { createPolicy, removePolicy, setPolicy } from "../acp/policy";
+import {
+  addForbiddenRuleUrl,
+  addOptionalRuleUrl,
+  addRequiredRuleUrl,
+  createRule,
+  Rule,
+} from "../acp/rule";
+import { acp } from "../constants";
+import {
+  IriString,
+  UrlString,
+  WithResourceInfo,
+  WithServerResourceInfo,
+} from "../interfaces";
 import { mockSolidDatasetFrom } from "../resource/mock";
-import { createSolidDataset } from "../resource/solidDataset";
-import { addIri } from "../thing/add";
-import { createThing, setThing } from "../thing/thing";
+import { addIri, addUrl } from "../thing/add";
+import { setThing } from "../thing/thing";
 import { internal_hasInaccessiblePolicies } from "./acp";
 
-type Policy = {
-  policies: string[];
-  memberPolicies: string[];
-  acrPolicies: string[];
-  memberAcrPolicies: string[];
+// Key: actor relation (e.g. agent), value: actor (e.g. a WebID)
+type MockRule = Record<UrlString, UrlString[]>;
```

Comment:
Would this be valid ?
```suggestion
type MockRule = Record<acp.agent | acp.group, UrlString[]>;
```

Reply:
Hmm good question - it might be. I'll give that a try.

Reply:
Very cool, it works: 6776b298

Now we get autocompletion when defining the mock object, which is very helpful given how complex they are. Thanks!